core-bundle/src/Twig/Loader/ContaoFilesystemLoader.php

```diff
@@ -105,23 +108,26 @@ public function getSourceContext(string $name): Source
             return new Source(file_get_contents($path), $templateName, $path);
         }
 
+        $getExtendedTemplate = static function ($path): string|null {
+            if (1 === preg_match('/\$this\s*->\s*extend\s*\(\s*[\'"]([a-z0-9_-]+)[\'"]\s*\)/i', (string) file_get_contents($path), $match)) {
+                return $match[1];
+            }
+
+            return null;
+        };
+
+        // Use the default path of the template if it extends itself
+        if (($extendedTemplate = $getExtendedTemplate($path)) && "@Contao/$extendedTemplate.html5" === $name) {
+            $this->framework->initialize();
+            $path = $this->framework->getAdapter(TemplateLoader::class)->getDefaultPath($extendedTemplate, 'html5');
+        }
+
         // Look up the blocks of the parent template if present
-        if (
-            1 === preg_match(
-                '/\$this\s*->\s*extend\s*\(\s*[\'"]([a-z0-9_-]+)[\'"]\s*\)/i',
-                (string) file_get_contents($path),
-                $match,
-            )
-            && '@Contao/'.$match[1].'.html5' !== $name
-        ) {
-            return new Source($this->getSourceContext('@Contao/'.$match[1].'.html5')->getCode(), $templateName, $path);
+        if (($extendedTemplate = $getExtendedTemplate($path)) && "@Contao/$extendedTemplate.html5" !== $name) {
```

Comment:
The `$extendedTemplate = $getExtendedTemplate($path)` part is the same as in line 120, isn‘t it? So theoretically we don‘t need an anonymous function that we call twice.

Reply:
It is not the same because the `$path` changes in between the two calls.
I also tried to prevent the need for the function, but I found no good way.
An alternative would be to make it a private method.

Reply:
No, let‘s keep it that way then. 👍